src/publishProject/projectUtils.ts

```diff
@@ -140,3 +159,41 @@ export function getPublishServerName(target: string) {
 export function validateSqlServerPortNumber(port: number): boolean {
     return Number.isInteger(port) && port >= 1 && port <= constants.MAX_PORT_NUMBER;
 }
+
+/**
+ * Retrieves and filters SQL Server container tags based on project target version.
+ * Used by Publish Project dialog to provide granular tag selection.
+ *
+ * @param targetVersion - The SQL Server version (e.g., "160" for SQL Server 2022)
+ * @returns Sorted array of tags filtered by version from deployment UI versions
+ */
+export async function getSqlServerContainerTagsForTargetVersion(
+    targetVersion?: string,
+): Promise<FormItemOptions[]> {
+    try {
+        // Get the deployment UI versions first
+        const deploymentVersions = await getSqlServerContainerVersions();
+        if (!deploymentVersions || deploymentVersions.length === 0) {
+            return [];
+        }
+
+        // Determine minimum year based on target version
+        let minYear = 2017; // Default to SQL Server 2017
```

Comment:
Why are we defaulting to such an old version instead of the latest GA?  FYI: there's a bug with 2025 on ARM that won't be fixed at GA.  Please sync with PMs and @laurenastrid1 about the solution here.

Reply:
In ADS we get max available year image tag incase no verison number is found from the project. Since we have current available images from 2017-2025 in deploymentUI, I used the min number 2017, but it seems what ADS is doing is right, we should provide the very latest image or all available images(checking with PM)